src/py/flwr/server/driver/inmemory_driver.py

```diff
@@ -142,7 +142,9 @@ def pull_messages(self, message_ids: Iterable[str]) -> Iterable[Message]:
         # Pull TaskRes
         task_res_list = self.state.get_task_res(task_ids=msg_ids)
         # Delete tasks in state
-        self.state.delete_tasks(msg_ids)
+        self.state.delete_tasks(
+            {UUID(task_res.task.ancestry[0]) for task_res in task_res_list}
+        )
```

Comment:
```suggestion
        # Delete the TaskIns/TaskRes pairs if TaskRes is found
        task_ins_ids_to_delete = {
            UUID(task_res.task.ancestry[0]) for task_res in task_res_list
        }
        self.state.delete_tasks(task_ins_ids=task_ins_ids_to_delete)
```

Should we make it consistent with the implementation in `ServerAppIoServicer`?